WooCommerce/Classes/Tools/UnitInputFormatter/ShippingInputFormatter.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+
+/// `UnitInputFormatter` implementation for shipping input.
+///
+struct ShippingInputFormatter: UnitInputFormatter {
+
+    // Match all the characters not included in the set: numbers between 0 and 9, and - . , %
+    let regex = "[^-0-9%.,]"
+
+    func isValid(input: String) -> Bool {
+        guard input.isEmpty == false else {
+            // Allows empty input to be replaced by 0.
+            return true
+        }
+
+        return input.range(of: regex, options: .regularExpression) == nil
+    }
+
+    func format(input text: String?) -> String {
+        guard let text = text, text.isEmpty == false else {
+            return "0"
+        }
+
+        let formattedText = text
+            // Removes leading 0s before the digits.
+            .replacingOccurrences(of: "^0+([1-9]+)", with: "$1", options: .regularExpression)
+            // Maximum one leading 0.
+            .replacingOccurrences(of: "^(0+)", with: "0", options: .regularExpression)
```

Comment:
It looks like we can merge these 2 replacements into one regex pattern: https://regexr.com/4t1pp. This new pattern is probably buggy though. 😄

Reply:
I am always inclined to divide the regex into several parts, to improve their readability (even if the performance may be lower).
In this case, as you said I think there is a problem with the regex since `^0+([^0.]|0.)` is matching also the first number after zeros.
![Schermata 2020-01-28 alle 13 31 42](https://user-images.githubusercontent.com/495617/73264227-8fd10800-41d2-11ea-94ea-a35f810f73d7.png)